packages/components/src/ComboBox/src/ComboBox.tsx

```diff
@@ -0,0 +1,337 @@
+    
+    const handleMouseDown: MouseEventHandler<HTMLElement> = useCallback(e => {
+        // If the input or button is the one that is clicked, we don't want to focus it since it's already done.
+        if (inputRef.current && e.target !== inputRef.current && buttonRef.current && e.target !== buttonRef.current) {
+            // forwards the focus to the input element when clicking on the input group.
+            inputRef.current.focus();
+            // This ensures that the input does not lose focus when clicking on the input group.
+            e.preventDefault();
+        }
+    }, [inputRef]);
+
+    const prefixMarkup = prefix ? (
+        <SlotProvider values={[
+            [TextContext, { size, className: styles["hop-ComboBox__prefix"] }],
+            [IconContext, { size, className: styles["hop-ComboBox__prefix"] }]
+        ]}
+        >
+            <ClearContainerSlots>
+                {isTextOnlyChildren(prefix) ? <Text>{prefix}</Text> : prefix}
+            </ClearContainerSlots>
+        </SlotProvider>
+    ) : null;
+
+    const footerMarkup = footer ? (
+        <Footer>{isTextOnlyChildren(footer) ? <Text>{footer}</Text> : footer}</Footer>
+    ) : null;
+
+    return (
+        <RACComboBox
+            {...otherProps}
+            ref={ref}
+            className={classNames}
+            style={style}
+            isInvalid={isInvalid}
+            menuTrigger={menuTrigger}
+        >
+            {comboBoxRenderProps => {
+                const { isOpen } = comboBoxRenderProps;
+                const ButtonIcon = isOpen ? AngleUpIcon : AngleDownIcon;
+
+                return (
+                    <>
+                        <SlotProvider values={[
+                            [LabelContext, {
+                                className: styles["hop-ComboBox__label"]
+                            }],
+                            [HelperMessageContext, {
+                                className: styles["hop-ComboBox__helper-message"]
+                            }],
+                            [ErrorMessageContext, {
+                                className: styles["hop-ComboBox__error-message"]
+                            }]
+                        ]}
+                        >
+                            {fieldChildren}
+                        </SlotProvider>
+                        <Group 
+                            {...otherTriggerProps}
+                            ref={triggerRef}
+                            className={triggerClassNames}
+                            style={triggerStyle}
+                            onMouseDown={handleMouseDown}
+                            data-selected={isOpen || undefined}
+                        >
+                            {prefixMarkup}
+                            <Input
+                                ref={mergedInputRefs}
+                                className={composeClassnameRenderProps(inputContext?.className, styles["hop-ComboBox__input"])}
```

Comment:
How do you guys usually decide to extract to as const or not?

Reply:
depends on how we feel that day. haha good question. My guess is it depends on length. I could extract this one if you feel it would be cleaner?

Reply:
Just wondering if we should set a convention, still very readable

Reply:
ya, I put it where I define the others.